test/Microsoft.AspNetCore.Mvc.Razor.Test/RazorViewEngineTest.cs

```diff
@@ -1710,18 +1677,26 @@ private RazorViewEngine CreateSuccessfulViewEngine()
 
         private TestableRazorViewEngine CreateViewEngine(
             IRazorPageFactoryProvider pageFactory = null,
-            IEnumerable<IViewLocationExpander> expanders = null)
+            IEnumerable<IViewLocationExpander> expanders = null,
+            IEnumerable<string> viewLocationFormats = null,
+            IEnumerable<string> areaViewLocationFormats = null)
         {
             pageFactory = pageFactory ?? Mock.Of<IRazorPageFactoryProvider>();
             return new TestableRazorViewEngine(
                 pageFactory,
-                GetOptionsAccessor(expanders));
+                GetOptions(expanders, viewLocationFormats, areaViewLocationFormats));
         }
 
-        private static IOptions<RazorViewEngineOptions> GetOptionsAccessor(
-            IEnumerable<IViewLocationExpander> expanders = null)
+        private static IOptions<RazorViewEngineOptions> GetOptions(
+            IEnumerable<IViewLocationExpander> expanders = null,
+            IEnumerable<string> viewLocationFormats = null,
+            IEnumerable<string> areaViewLocationFormats = null)
         {
+            var optionsSetup = new RazorViewEngineOptionsSetup(Mock.Of<IHostingEnvironment>());
```

Comment:
Specifically what I'm suggesting is forgetting about `RazorViewEngineOptionsSetup` here. Instead,
``` c#
if (viewLocationFormats == null)
{
    // Do tests that don't pass in formats even care what the values are? If not
    options.ViewLocationFormats.Add("nowhere");
    // If they do, ...
    options.ViewLocationFormats.Add("/Views/{1}/{0}.cshtml");
    options.ViewLocationFormats.Add("/Views/Shared/{0}.cshtml");
}
else
{
     foreach (var location in viewLocationFormats)
     {
         options.ViewLocationFormats.Add(location);
     }
}
```